src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1261,22 +1267,6 @@
   {:pre [(kitchensink/datetime? time)]}
   (jdbc/delete! :reports ["producer_timestamp < ?" (to-timestamp time)]))
 
-(defn delete-old-unassociated-catalogs!
-  [time]
-  {:pre [(kitchensink/datetime? time)]}
-  (when @store-catalogs-historically?
-    (let [time (to-timestamp time)]
-      (jdbc/do-prepared
-       (str "DELETE FROM catalogs oc "
-            "       WHERE oc.producer_timestamp < ? "
-            "       AND NOT EXISTS (SELECT 1 FROM catalogs c "
-            "                       JOIN reports r ON c.catalog_uuid = r.catalog_uuid "
```

Comment:
I just figured we don't really need the burden of the GC task anymore, but I can add it back if you guys think it's necessary.